test/user_repository_test.js

```diff
@@ -1,11 +1,15 @@
+'use strict'
 const assert = require('assert')
 const userRepository = require('../lib/user_repository')
 
 describe('UserRepository', function() {
-  it('returns user object on given username', function(done) {
-    user = userRepository.findFromUsername('testuser', done)
-
-    assert.equal('test_token', user.token)
-    assert.equal('test_project', user.project)
+  it('returns user object on given username', function() {
+    this.timeout(5000)
+    return userRepository.findFromUsername('testuser')
```

Comment:
use mocha `Promise`-type tests by returning a promise in the test case.
it times out if the Promise hasn't been yet resolved. You can attach to the chain and make assertions on the result (passed as argument)